src/idlcxx/src/generator.h

```diff
@@ -17,26 +17,33 @@
 struct generator {
   const char *path;
   char *array_format;
+  const char *array_swap;
   const char *array_include;
   char *sequence_format;
+  const char *sequence_swap;
   const char *sequence_include;
   char *bounded_sequence_format;
+  const char *bounded_sequence_swap;
   const char *bounded_sequence_include;
   char *string_format;
+  const char *string_swap;
   const char *string_include;
   char *bounded_string_format;
+  const char *bounded_string_swap;
   const char *bounded_string_include;
   char *union_format;
   char *union_getter_format;
+  const char *union_swap;
   const char *union_include;
-  char *swap_format;
-  const char *swap_include;
+  const char *basic_swap;
+  const char *basic_swap_include;
   bool uses_array;
   bool uses_sequence;
   bool uses_bounded_sequence;
   bool uses_string;
   bool uses_bounded_string;
   bool uses_union;
+  bool uses_basic_swap;
```

Comment:
If we have a file which only constructs types out of types imported from other files, than this file does not need the basic_swap functions, since the basic type swap functions will be included from those files.

I will move the swap implementation to its own PR.